client/src/components/App/App.js

```diff
@@ -21,19 +21,47 @@ class App extends Component {
       newUsername: "",
       newEmail: "",
       newPassword: "",
-      newPasswordConfirm: ""
-
+      newPasswordConfirm: "",
+      signupError: "",
+      disabledProp: true
     };
     this.handleChange = this.handleChange.bind(this);
     this.handleLogin = this.handleLogin.bind(this);
     this.handleRegister = this.handleRegister.bind(this);
   }
 
   handleChange(event) {
-
     const { name } = event.target;
     const { value } = event.target;
     this.setState({ [name]: value });
+    let errorMessage = "";
+    let disabled = true;
+    if (
+      this.state.newUsername === "" &&
+      this.state.newPassword === "" &&
+      this.state.newEmail === ""
+    ) {
+      errorMessage = "";
+    } else if (!this.state.newUsername.match(/[0-9a-zA-z]{6,20}/g)) {
+      errorMessage =
+        "Username must contain only numbers and letters with length between 6 and 20";
+    } else if (
+      !/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/.test(this.state.newEmail)
+    ) {
+      errorMessage = "Must use a valid email address";
+    } else if (!this.state.newPassword.match(/[0-9a-zA-z]{6,20}/g)) {
```

Comment:
should we add like a minimum length for the password?

Reply:
The min is 6 and the max 20 '{6,20}'